kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyInputFormat.java

```diff
@@ -137,6 +139,12 @@ public ConfigBuilder readFrom(String uri) {
       conf.setClass(KITE_TYPE, type, type);
       return this;
     }
+
+    public ConfigBuilder withSchema(Schema readerSchema) {
```

Comment:
I'd keep this, but add validation to make sure the class isn't set by a view. Also, the methods that take views should call it where appropriate. And possibly add `select(String...)`?